app/utils/dateConverter.js

```diff
@@ -1,21 +1,33 @@
 const twelveTotwenty = function(twelve) {
     
-  var strToNum = twelve.length > 4 ? +twelve.substr(0, 2) : +twelve.substr(0, 1);
-  var a = strToNum + 12 == 24 ? a = '00' : strToNum + 12; 
-  var b = strToNum > 9 ? twelve.substr(2, 3) : twelve.substr(1, 3);
+  var strToNum = twelve.length > 6 ? +twelve.substr(0, 2) : +twelve.substr(0, 1);
+  var AmCheck = twelve.slice(-2).toLowerCase();
+  var PmCheck = twelve.slice(-2).toLowerCase();
+  var a = strToNum + 12 == 24 ? a = '00' : AmCheck == 'am' ? strToNum = '0' + strToNum : strToNum + 12; 
+  var b = strToNum > 9 ? twelve.substr(2, 3) : (strToNum <= 9 && twelve.length > 4) ? twelve.substr(2, 3) : twelve.substr(1, 3);
   var twenty = a + b; 
-  
+
+    if (strToNum > 24 || twelve.length == 5) {
```

Comment:
Record like `if (strToNum > 24 || twelve.length == 5) twenty = null;` will be shorter. Depends on you